From dfc1107016190caf4835953a5f930cf6500765ae Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Tue, 29 Apr 2014 08:19:51 -0400
Subject: [net] fib_frontend: fix possible NULL pointer dereference

Message-id: <fc1f2871674ee258ad1ef3d0a1b2762ea0edbfab.1398759582.git.jbenc@redhat.com>
Patchwork-id: 79220
O-Subject: [RHEL7.1 net PATCH] fib_frontend: fix possible NULL pointer dereference
Bugzilla: 1091142
RH-Acked-by: Veaceslav Falico <vfalico@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1091142
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7393292

Tested: using the steps to reproduce described in bugzilla

commit a0065f266a9b5d51575535a25c15ccbeed9a9966
Author: Oliver Hartkopp <socketcan@hartkopp.net>
Date:   Thu Jan 23 10:19:34 2014 +0100

    fib_frontend: fix possible NULL pointer dereference

    The two commits 0115e8e30d (net: remove delay at device dismantle) and
    748e2d9396a (net: reinstate rtnl in call_netdevice_notifiers()) silently
    removed a NULL pointer check for in_dev since Linux 3.7.

    This patch re-introduces this check as it causes crashing the kernel when
    setting small mtu values on non-ip capable netdevices.

    Signed-off-by: Oliver Hartkopp <socketcan@hartkopp.net>
    Acked-by: Eric Dumazet <edumazet@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/fib_frontend.c b/net/ipv4/fib_frontend.c
index f3269a0..5182fc7 100644
--- a/net/ipv4/fib_frontend.c
+++ b/net/ipv4/fib_frontend.c
@@ -1049,6 +1049,8 @@ static int fib_netdev_event(struct notifier_block *this, unsigned long event, vo
  }
 
  in_dev = __in_dev_get_rtnl(dev);
+ if (!in_dev)
+  return NOTIFY_DONE;
 
  switch (event) {
  case NETDEV_UP:
-- 
1.7.1